Lib/test/test_posix.py

```diff
@@ -1794,6 +1794,39 @@ def test_dup2(self):
         with open(dupfile) as f:
             self.assertEqual(f.read(), 'hello')
 
+    @unittest.skipUnless(hasattr(os, 'posix_spawnp'), "test needs os.posix_spawnp")
+    def test_posix_spawnp(self):
+        # Use a symlink to create a program in its own temporary directory
+        temp_dir = tempfile.mkdtemp()
+        self.addCleanup(support.rmtree, temp_dir)
+
+        program = 'posix_spawnp_test_program.exe'
+        program_fullpath = os.path.join(temp_dir, program)
+        os.symlink(sys.executable, program_fullpath)
+
+        path = os.environ.get('PATH', '')
+        path = os.pathsep.join((temp_dir, path))
```

Comment:
nitpick: maybe use the following code if PATH is unset

```
    try:
        path = os.pathsep.join((temp_dir, os.environ['PATH']))
    except KeyError:
        path = temp_dir   # PATH is not set
```

I'm not sure of the behavior if you add an empty string to PATH, like PATH="/path/to/test:".